hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -110,12 +108,8 @@ long getLastCallId() {
      */
     public boolean register(Invocation invocation) {
         final long callId;
-        try {
-            boolean force = invocation.op.isUrgent() || invocation.isRetryCandidate();
-            callId = force ? callIdSequence.forceNext() : callIdSequence.next();
-        } catch (TimeoutException e) {
-            throw new HazelcastOverloadException("Failed to start invocation due to overload: " + invocation, e);
-        }
+        boolean force = invocation.op.isUrgent() || invocation.isRetryCandidate();
```

Comment:
This is a problem because you loose the information about the invocation in the exception.

Reply:
Changing to put the Invocation in signature of next API (i.e. next(Invocation)). Does not look very good but it will also make the client and server side usage both have the invocation information inside the thrown exception.